api/src/test/java/org/openmrs/api/ConceptServiceTest.java

```diff
@@ -3531,4 +3531,38 @@ public void getConcepts_shouldPassWithAndOrNotWords() {
 		Assert.assertEquals(1, searchResults.size());
 		assertThat(searchResults.get(0).getWord(), is("SALBUTAMOL INHALER NOT"));
 	}
+	
+	/**
+	 * @see ConceptService#getConceptByNameOrIdOrMap(String)
+	 */
+	@Test
+	public void getConceptByNameOrIdOrMap_shouldReturnNullGivenNullParameter() {
+		Assert.assertNull(Context.getConceptService().getConceptByNameOrIdOrMap(null));
+	}
+
+	/**
+	 * @see ConceptService#getConceptByNameOrIdOrMap(String)
+	 */
+	@Test
+	public void getConceptByNameOrIdOrMap_shouldReturnNullGivenWrongInput() {
+		assertNull(conceptService.getConceptByNameOrIdOrMap("Some wrong input that matches no concept"));
+	}
+
+	/**
+	 * @see ConceptService#getConceptByNameOrIdOrMap(String)
+	 */
+	@Test
+	public void getConceptByNameOrIdOrMap_shouldGetConceptByItsMap() {
+		Concept concept = conceptService.getConceptByNameOrIdOrMap("WGT:234");
+		Assert.assertEquals(5089, concept.getId().intValue());
```

Comment:
I had recently gone through the test datasets and there the code for a concept_reference_term was "WGT234" and not "WGT:234". I think here your returned concept is null because of "WGT:234". Your tests might be failing because of it.